web/permissions/permissions_test.go

```diff
@@ -681,6 +708,31 @@ func TestCannotFindToken(t *testing.T) {
 	assert.Contains(t, err.Error(), "Cannot find token for shortcode")
 }
 
+func TestTinyShortCodeOK(t *testing.T) {
+	id, codes, _ := createTestTinyCode(token, "elise", "30m")
+	code := codes["elise"]
+	assert.Len(t, code, 6)
+
+	perm, _ := permission.GetByID(testInstance, id)
+	assert.Equal(t, code, perm.ShortCodes["elise"])
+
+	assert.NotNil(t, perm.ShortCodes)
+
+	req1, _ := http.NewRequest("GET", ts.URL+"/permissions/self", nil)
+	req1.Header.Add("Authorization", "Bearer "+perm.ShortCodes["elise"])
+	res1, _ := http.DefaultClient.Do(req1)
+	assert.Equal(t, res1.StatusCode, http.StatusOK)
+
+	token, _ := permission.GetTokenFromShortcode(testInstance, perm.ShortCodes["elise"])
+	assert.Equal(t, perm.Codes["elise"], token)
+}
+
+func TestTinyShortCodeInvalid(t *testing.T) {
+	_, codes, _ := createTestTinyCode(token, "fanny", "24h")
+	code := codes["fanny"]
+	assert.Len(t, code, 12)
```

Comment:
In fact, I'm wondering if we should not drop the shortcode creation (by returning a 400 error code) if tiny is set and duration > 1h ?

Reply:
I don't think so. I have hesitated when coding it, but I think we may want to change the limit and it should not break the apps: they will have a shortcode instead of a tinycode but it should not break anything as the format of the response is the same.